design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+
+![](/assets/design/specs/add-formula-column/12.png)
+
+### Scenario 2 - Adding an erroneous parameter to the formula 
+
+Parameters of different formulas accept columns of different data types and values of different data types. The erroneous parameter here means the data type of the entered value or column is incompatible with the formula. 
+
+For instance, the length formula expects a list value or a list type column as an input to the parameter `List` but it gets a text type column which is movieNames and hence throws an error and the user is not able to see a filled column.
+
+![](/assets/design/specs/add-formula-column/14.png)
+
+### Scenario 3 - Edit a formula column 
+
+#### Scenario 3.a - Edit a formula before adding the column
+
+Before adding the column, user can edit the formula with no constraints. 
+
+Case 1 - Edit parameters before adding column.
+Parameters can be easily edited at this point. To not run into errors, the users would just have to take care of the data type compatibility.
+
+Case 2 - Change the formula before adding the column
+This can also be easily done. The user just have to select another formula from the selection list. Then they will be able to see the parameters relevant to the selected formula and they will be empty. 
+
+#### Scenario 3.b - Edit a formula after adding the column
+
+Users can only change the parameters after adding the column to the query. Again users might run into errors if they enter incompatible values/columns. More details TDB. 
+
+### Scenario 4 - Click outside the inspector
+
+#### Scenario 4.a - Clicks outside before adding name or selecting formula
+
+If the users click outside the inspector before even touching the formula settings that is adding name or selecting formula, there won't be any empty column added. And there would be absolutely no change in the explorer.
+
+#### Scenario 4.b - Clicks outside after adding name or selecting formula
+
+If there is any activity in the formula settings that is if the name of the column is added or the formula is selected, an empty column will be added. 
+
+Once users click outside, they will see a warning on the column so users know there is something left to do with the formula column. 
+
+![](/assets/design/specs/add-formula-column/15.png)
+
+#### Scenario 4.c - Clicks outside after successfully adding the formula
+
+If the formula and its parameters are added successfully and the preview column is filled. When the user clicks outside the preview is still visible. The user will get the inspector view of formula back if they click on the column. 
+
+![](/assets/design/specs/add-formula-column/16.png)
+
+#### Scenario 4.d - Clicks outside after adding erroneous parameters
+
+If the users have added erroneous parameters then they click outside the inspector, the parameter values will be lost and they will see a similar empty column preview with a warning. 
```

Comment:
I think the process should go as follows:
- The user adds a formula column, and the inspector displays the column properties (name) and the formula selector.
- The table displays a preview of the column with no data.
- The user selects a formula, and a list of formula parameters is displayed.
- The user enters formula parameters.
- Once they select a column, we can begin displaying the output in the preview.
- The user can deselect a formula column at any time, and this should not result in the column being removed.

I believe we must ensure that validation in the formula form prevents invalid values from being entered.